src/blog/2024/04/building-an-admin-panel-in-node-red-with-dashboard-2.md

```diff
@@ -0,0 +1,104 @@
+2. Drag an event node on the canvas, then click on it, and select the UI base that contains your all pages including the admin page 
+3. Drag a switch node on the canvas, and add two conditions, one to check whether the user’s username is contained in the admin list or a second for otherwise, see the below image.
+
+!["Screenshot displaying the switch node which checks whether the logged-in user's username is contained in the admin list or not"](./images/building-admin-panel-node-red-dashboard-2-switch-node-checking-page-viewer-isadmin.png "Screenshot displaying the switch node which checks whether the logged-in user's username is contained in the admin list or not"){data-zoomable}
+
+4. Drag two change nodes onto the canvas, Configure the first change node to show the admin page by setting `msg.payload` as `{"pages":{"show":["Admin View"]}}`, and the second change node to hide the admin page by setting the payload as: `{"pages":{"hide":["Admin View"]}}`.
+
+!["Screenshot displaying the change node which contains payload to show admin page"](./images/building-admin-panel-node-red-dashboard-2-change-node-for-showing-page.png "Screenshot displaying the change node which contains payload to show admin page"){data-zoomable}
+
+!["Screenshot displaying the change node which contains payload to hide admin page"](./images/building-admin-panel-node-red-dashboard-2-change-node-for-hidding-page.png "Screenshot displaying the change node which contains payload to display admin page"){data-zoomable}
+
+5. Connect the first change node's input to the switch node's first output and the second change node's input to the switch node's second output.
+6. Drag a ui-control widget onto the canvas, then click on it and select ui-base which includes all your pages including the admin page. 
+7. Finally, connect both change node’s outputs to the ui-control’s input.
+
+## Hidding Admin only page by default 
+
+finally, we need to hide an admin-only page by default to ensure regular users don't accidentally land on the admin-only page.
+
+1. Go to the Dashboard 2.0 sidebar, and select the layout tab.
+2. Locate the admin-only page and click on the edit icon next to it.
+3. Set visibility as "hidden".
+
+!["Screenshot displaying admin-only page configuration"](./images/building-admin-panel-node-red-dashboard-2-admin-only-page-configuration.png "Screenshot displaying admin-only page configuration"){data-zoomable}
+
+## Deploying the flow
+
+!["Screenshot displaying the FlowFuse Editor with flow of admin-only page"](./images/building-admin-panel-node-red-dashboard-2-flowfuse-editior.png "Screenshot displaying the FlowFuse Editor with flow of admin-only page"){data-zoomable}
+
+1. With your flow updated to include the above, click the "Deploy" button in the top-right of the Node-RED Editor.
+2. Navigate to `https://<your-instance-name>.flowfuse.cloud/dashboard`.
```

Comment:
Opening the dashboard through the sidebar might be a better option than this line for the next post @sumitshinde-84